### PR TITLE
Edit

### PR DESCRIPTION
**Pull request by wunderboyy**
I replaced all the `var` with `const` or `let`

### CODE REVIEW

**Comment by hamishwillee** [comment]
Thanks but I can't merge this (at least yet). The tutorial has an accompanying demo app that must be kept in sync https://github.com/mdn/express-locallibrary-tutorial

**Comment by hamishwillee** [comment]
Thank you. How did you test the changes?

**Comment by kyopa** [comment]
I tried it on my computer and it worked just like before.

**Comment by hamishwillee** [comment]
Thank you. I will try do a review later this week.

File: controllers/authorController.js [hunk 1]
```diff
@@ -65,7 +65,7 @@ exports.author_create_post = [
         const errors = validationResult(req);
         
         // Create Author object with escaped and trimmed data
-        var author = new Author(
+        ' author = new Author(
```
Comment (hamishwillee) [inline controllers/authorController.js:68]: Error. 
```suggestion
        const author = new Author(
```

File: controllers/authorController.js [hunk 2]
```diff
@@ -32,7 +32,7 @@ exports.author_detail = function (req, res, next) {
     }, function (err, results) {
         if (err) { return next(err); } // Error in API usage.
         if (results.author == null) { // No results.
-            var err = new Error('Author not found');
+            let err = new Error('Author not found');
```
Comment (hamishwillee) [inline controllers/authorController.js:35]: Shouldn't this be a `const`?

File: controllers/authorController.js [hunk 3]
```diff
@@ -152,7 +152,7 @@ exports.author_update_get = function (req, res, next) {
     Author.findById(req.params.id, function (err, author) {
         if (err) { return next(err); }
         if (author == null) { // No results.
-            var err = new Error('Author not found');
+            let err = new Error('Author not found');
```
Comment (hamishwillee) [inline controllers/authorController.js:155]: Should this be a `const`?

File: controllers/bookController.js [hunk 4]
```diff
@@ -65,7 +65,7 @@ exports.book_detail = function(req, res, next) {
     }, function(err, results) {
         if (err) { return next(err); }
         if (results.book==null) { // No results.
-            var err = new Error('Book not found');
+            let err = new Error('Book not found');
```
Comment (hamishwillee) [inline controllers/bookController.js:68]: Should this be a `const`? I mean, I don't think that err is reassigned after this point?

File: controllers/bookController.js [hunk 5]
```diff
@@ -236,14 +236,14 @@ exports.book_update_get = function(req, res, next) {
         }, function(err, results) {
             if (err) { return next(err); }
             if (results.book==null) { // No results.
-                var err = new Error('Book not found');
+                let err = new Error('Book not found');
```
Comment (hamishwillee) [inline controllers/bookController.js:239]: Should this be a `const`?

File: controllers/bookinstanceController.js [hunk 6]
```diff
@@ -25,7 +25,7 @@ exports.bookinstance_detail = function(req, res, next) {
     .exec(function (err, bookinstance) {
       if (err) { return next(err); }
       if (bookinstance==null) { // No results.
-          var err = new Error('Book copy not found');
+          let err = new Error('Book copy not found');
```
Comment (hamishwillee) [inline controllers/bookinstanceController.js:28]: Should this be a `const`?

File: controllers/bookinstanceController.js [hunk 7]
```diff
@@ -137,7 +137,7 @@ exports.bookinstance_update_get = function(req, res, next) {
         }, function(err, results) {
             if (err) { return next(err); }
             if (results.bookinstance==null) { // No results.
-                var err = new Error('Book copy not found');
+                let err = new Error('Book copy not found');
```
Comment (hamishwillee) [inline controllers/bookinstanceController.js:140]: Should this be a `const`?

File: controllers/genreController.js [hunk 8]
```diff
@@ -35,7 +35,7 @@ exports.genre_detail = function(req, res, next) {
     }, function(err, results) {
         if (err) { return next(err); }
         if (results.genre==null) { // No results.
-            var err = new Error('Genre not found');
+            let err = new Error('Genre not found');
```
Comment (hamishwillee) [inline controllers/genreController.js:38]: Should this be a `const`?

File: controllers/genreController.js [hunk 9]
```diff
@@ -157,7 +157,7 @@ exports.genre_update_get = function(req, res, next) {
     Genre.findById(req.params.id, function(err, genre) {
         if (err) { return next(err); }
         if (genre==null) { // No results.
-            var err = new Error('Genre not found');
+            let err = new Error('Genre not found');
```
Comment (hamishwillee) [inline controllers/genreController.js:160]: Should this be a `const`?

File: populatedb.js [hunk 10]
```diff
@@ -3,38 +3,38 @@
...
 db.on('error', console.error.bind(console, 'MongoDB connection error:'));
 
-var authors = []
-var genres = []
-var books = []
-var bookinstances = []
+let authors = []
+let genres = []
+let books = []
+let bookinstances = []
```
Comment (hamishwillee) [inline populatedb.js:30]: Can all of these be `const`? Yes we push values to the arrays, but I don't *think* we reassign the array itself?

**Pull request closed without merging**